cargo-ext/src/new_project.rs

```diff
@@ -0,0 +1,41 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use cargo_generate::GenerateArgs;
+use cargo_generate::TemplatePath;
+
+pub fn new_project(path: PathBuf) -> anyhow::Result<()> {
+    let name = path
+        .file_name()
+        .ok_or_else(|| {
+            anyhow::anyhow!("Failed to get the last segment of the provided path for the project name")
+        })?
+        .to_str()
+        .ok_or_else(|| {
+            anyhow::anyhow!("The last segment of the provided path must be valid UTF8 to generate a valid project name")
+        })?
+        .to_string();
+
+    let generate_args = GenerateArgs {
+        template_path: TemplatePath {
+            git: Some("https://github.com/0xPolygonMiden/rust-templates".into()),
+            auto_path: Some("library".into()),
+            ..Default::default()
+        },
+        destination: path
+            .parent()
+            .map(|p| {
+                use path_absolutize::Absolutize;
```

Comment:
I don't think you need a library for this, `path.canonicalize()` (or alternatively `std::fs::canonicalize`) produce a canonical, absolute path.

Reply:
Good catch! Thanks! Fixed.